ush/retrieve_data.py

```diff
@@ -737,27 +756,40 @@ def main(argv):
     """
 
     cla = parse_args(argv)
+    if cla.cycle_date is None:
+        cla.cycle_date = to_datetime("1999123100") # set it to something benign
+    if cla.fcst_hrs is None:
+        cla.fcst_hrs = [0]                         # set a default
     cla.fcst_hrs = arg_list_to_range(cla.fcst_hrs)
 
     if cla.members:
         cla.members = arg_list_to_range(cla.members)
 
     setup_logging(cla.debug)
-    print("Running script retrieve_data.py with args:\n", f"{('-' * 80)}\n{('-' * 80)}")
+    print("Running script retrieve_data.py with args:", f"\n{('-' * 80)}\n{('-' * 80)}")
     for name, val in cla.__dict__.items():
         if name not in ["config"]:
             print(f"{name:>15s}: {val}")
     print(f"{('-' * 80)}\n{('-' * 80)}")
 
+    # with no --config arg passed, use our default config specs
+    if cla.config is None:
+        logging.info(f"read parm/data_locations.yml")
+        cla.config = config_exists("./parm/data_locations.yml")
+    logging.debug(f"config {cla.config}")
+
     if "disk" in cla.data_stores:
         # Make sure a path was provided.
         if not cla.input_file_path:
-            raise argparse.ArgumentTypeError(
-                (
-                    "You must provide an input_file_path when choosing "
-                    " disk as a data store!"
+            # See if file_path is in the config ...
+            file_path = cla.config.get(cla.external_model, {}).get("disk", {}).get("file_path")
+            if not file_path:
+                raise argparse.ArgumentTypeError(
+                    (
+                        "You must provide an input_file_path when choosing "
+                        " disk as a data store!"
+                    )
```

Comment:
Please do stick with this one being a required argument for a similar reasons I've mentioned above.

Reply:
done.

Reply:
This logic should be reset to the original.